shellscape/webpack-plugin-serve
# Reproduction test case for issue #89

This PR contains:

- [ ] metadata

### Breaking Changes?


### Please Describe Your Changes

This is some sample code to reproduce the issue I described in https://github.com/shellscape/webpack-plugin-serve/issues/89.

I've added a new `recipe` called `production` which is a copy of the `react` recipe with the following changes:

- `npm run start` now runs the app with the `production` environment variable
- `webpack.config.js` is now set to `mode: production`
- `webpack.config.js` now includes the following section to generate a vendor bundle for all `node_modules`. This was added to deliberately slow down the builds and ensure a dedicated bundle is created so it's easier to see the problem:

```
optimization: {
    splitChunks: {
      cacheGroups: {
        vendor: {
          chunks: 'all',
          name: 'vendor',
          test: /[\\/]node_modules[\\/]/
        }
      }
    }
  }
```

To reproduce the issue use these steps:

- Start the server with `npm run start` (or `yarn start`)
- Note that a full production build occurs including a dedicated `1.bundle.js` which contains the node_modules bundle
- Make a change to the App.js file and save to trigger a HMR update
- Note that webpack rebuilds ALL bundles including the vendor bundle, rather than only the affected bundle

Here's a GIF illustrating the steps:

![repro](https://user-images.githubusercontent.com/226640/51050927-2f4f3880-1587-11e9-848b-a72cfb4c099c.gif)

In this example the issue isn't a massive problem because the vendor bundle is only about 1mb and rebuilds quite quickly, but in our case the vendor bundle(s) are massive (20mb+) so a full rebuild takes nearly a full minute which means trying to work with the dev servers is very difficult.
